third_party/move/move-bytecode-verifier/src/struct_defs.rs

```diff
@@ -91,8 +92,17 @@ impl<'a> StructDefGraphBuilder<'a> {
         let struct_def = StructDefinitionView::new(self.module, struct_def);
         // The fields iterator is an option in the case of native structs. Flatten makes an empty
         // iterator for that case
-        for field in struct_def.fields().into_iter().flatten() {
-            self.add_signature_token(neighbors, idx, field.signature_token(), false)?
+        let variant_count = struct_def.variant_count();
+        if variant_count > 0 {
+            for i in 0..variant_count {
+                for field in struct_def.fields_optional_variant(Some(i as VariantIndex)) {
+                    self.add_signature_token(neighbors, idx, field.signature_token(), false)?
+                }
+            }
+        } else {
+            for field in struct_def.fields().into_iter().flatten() {
```

Comment:
nit: `fields_optional_variant(None)` for explicit non-enum and remove `fields` completely, as its very error prone?

Reply:
done